docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,509 @@
+
+### Implementation specs
+
+#### initialization
+
+After dynamically generating the random symmetric key and encrypting the image layer, 
+two special JSON structures need to be generated, One is used to record public encrypted information 
+(such as encryption algorithm), 
+which is called [`PublicLayerBlockCipherOptions`](https://github.com/containers/ocicrypt-rs/blob/9c8691379012d666f210c1ef0f2a8bbf8caee53d/src/blockcipher/mod.rs#L77), and the other is used to record secret information 
+(such as symmetric key used for encryption), which is called [`PrivateLayerBlockCipherOptions`](https://github.com/containers/ocicrypt-rs/blob/9c8691379012d666f210c1ef0f2a8bbf8caee53d/src/blockcipher/mod.rs#L53). 
+The example are as follows:
+
+- `PublicLayerBlockCipherOptions`:
+
+  ```json
+  {
+      "cipher": "AES_256_CTR_HMAC_SHA256",
+      "hmac": “M0M5OTA5QUZFQzI1MzU0RDU1MURBR…”
+      "cipheroptions": {}
+  }
+  ```
+
+- `PrivateLayerBlockCipherOptions`:
+
+  ```json
+  {
+      "symkey": "54kiln1USEaKnlYhKdz+aA==",
+      "cipheroptions": {
+          "nonce": "AdcRPTAEhXx6uwuYcOquNA=="
+      }
+  }
+  ```
+ 
+#### Key Wrapping
+
+The `PrivateLayerBlockCipherOptions` needs to be encrypted through the [key provider protocol](https://github.com/containers/ocicrypt/blob/main/docs/keyprovider.md) with a "key provider program", 
+this action is usually called "Key Wrapping".
+the key provider program here also allows the owner to customize the implementation,
+as long as it meets the implementation specs described here.
+
+The key provider program should encrypt the `PrivateLayerBlockCipherOptions` with the owner's key.
+After that, it should Base64 encode the encrypted `PrivateLayerBlockCipherOptions` and then package it into a owner customized JSON structure. 
+In the confidential container design, this JOSN structure is called "annotation packet", it record some additional information, 
+such as the ID of the owner's key, for example:
+
+```
+{
+    "key_id":"1234",
+    "wrapped_data":#base64encode(Enc(PrivateLayerBlockCipherOptions)),
+    "wrap_type":"",
+}
+```
+
+The specific format of the annotation packet can be determined by the customized key provider program used by the owner,
+but it is necessary to ensure that the key provider program used by the decryptor 
+(i.e. the attestation-agent integrated with the specified [Key Broker Client](https://github.com/confidential-containers/attestation-agent/blob/main/docs/IMPLEMENTATION.md#kbc)) 
+supports parsing annotation packets of the same format.
+(In CCv1, when decrypting the image, the decryption of `PrivateLayerBlockCipherOptions` will be carried out by ocicrypt-rs through the [key provider protocol with the attestation-agent](https://github.com/confidential-containers/attestation-agent/blob/main/docs/IMPLEMENTATION.md#keyprovider-protocol) (as a key provider program).)
+
+#### Update manifest
+
+The manifest of container image provides two fileds in layer annotation 
```

Comment:
fileds -> fields

Reply:
"The manifest of container image" -> "The container image manifest"

Reply:
What do you mean with "layer annotation" ? I see no annotation in https://github.com/opencontainers/image-spec/blob/main/layer.md, and no layers in https://github.com/opencontainers/image-spec/blob/main/annotations.md. When I search for it, I get https://supercontainers.github.io/containers-wg/ideas/layer-annotation/, but that does not seem to match what you are talking about. So should that just be "annotations"?